mink/tasks/constraints_task.py

```diff
@@ -0,0 +1,145 @@
+"""Constraints task implementation."""
+
+from __future__ import annotations
+
+from typing import Optional
+
+import numpy as np
+import numpy.typing as npt
+import mujoco
+
+from ..configuration import Configuration
+from .task import Task
+from .exceptions import TaskDefinitionError
+
+
+class ConstraintsTask(Task):
+    """Regulate the equality constraints in the MuJoCo model.
+
+    This task aims to satisfy the equality constraints defined in the MuJoCo model.
+    It extracts the constraint error and Jacobian directly from the MuJoCo data
+    structure.
+
+    Attributes:
+        constraint_cost: Cost vector or scalar for the constraint error.
+    """
+
+    def __init__(
+        self,
```

Comment:
What do you think about having model be required in the constructor and then throwing an error if `model.neq` is 0?